### PR TITLE
Core vendor fix 1

### PR DESCRIPTION
**Pull request by AaronDeeds**
I noticed when serving vendor assets it wasn't checking if the file exists before trying to serve it. I also figured it should be in a function similar to the module assets.

### CODE REVIEW

File: engine/Core.php [hunk 1]
```diff
@@ -8,14 +8,22 @@ class Core {
...
 
-            $vendor_file_path = explode('vendor/', ASSUMED_URL)[1];
-            $vendor_file_path = '../vendor/'.$vendor_file_path;
+    private function serve_vendor_asset() {
+        $vendor_file_path = explode('/vendor/', ASSUMED_URL)[1];
+        $vendor_file_path = '../vendor/'.$vendor_file_path;
+        if (file_exists($vendor_file_path)) {
             $content_type = mime_content_type($vendor_file_path);
-
+            die($vendor_file_path);
```
Comment (delboy1978uk) [inline engine/Core.php:26]: you forgot to remove this?

Comment (AaronDeeds) [reply engine/Core.php:26]: Yes! Thank you, I didnt notice I left it in there after testing.

Comment (davidjconnelly) [reply engine/Core.php:26]: I must confess, I don't entirely understand the pull request.  

Please don't assume that I'm being smug or critical when I say this.  I'm not.  My head is elsewhere just now and I'm having a hard time wrapping my mind around this one.  Reach out to me if you'd like to discuss the pull request further.  I very much appreciate you taking the time to look at the code and submit your pull request.  Thank you!  And thanks to Derek, too!

**Comment by AaronDeeds** [comment]
I was interested to learn the execution flow of the framework and as I was
following the flow I seen some things I would change before using it. I
figured I could just change things to my liking and have a private version
of Trongate or I could do a simple pull request to see how receptive you
were to outside input.

Anyway, I noticed that the core throws hard php errors when using any URL
with "vendor" in the path and assumed it was done in a rush so I fixed it
and put it into its own function.

http://localhost/ANYTHING/vendor/ANYTHING

This also seems to unintentionally reserve the word "vendor" from being
used as a controller or method name.

**Pull request closed without merging**